pages/assets/js/src/profile-pages/pg-dashboard.ts

```diff
@@ -158,4 +170,66 @@ export class PgDashboard extends OpenElement {
       </div>
     `;
   }
+
+  private async getLocationFromIP() {
+    const saved_location = localStorage.getItem("user_location");
+    this.user.location = saved_location ? JSON.parse(saved_location) : null;
+    if (this.user.location?.hash) {
+      this.user.location_hash = this.user.location.hash;
+    }
+
+    if (
+      !this.user.location ||
+      (this.user.location.date_set &&
+        this.user.location.date_set <
+        Date.now() - 604800000) /*7 days in milliseconds*/
```

Comment:
every 7 days refresh the location is a nice idea

Reply:
although thinking about the source = 'ip' thing, should this condition also check if the user has manually set their location? so if source = 'user' then don't refresh it?